### DISCUSSION TITLE
Plugin lgthing actuel et Docker

### DISCUSSION THREAD

**Discussion by paifre** [pifou25/jeedom-lgthinq-plugin#48] [General]
Bonjour,

J'utilise votre plugin. Je tiens à vous remercier pour le travail accompli et j'ai vu que la v2 est en cours, top
Suite au dernier update, le plugin n'arrive plus à résoudre les dépendances Python, même en beta
Analysant les dépendances, j'ai vu un embryon de Docker
A partir de là, utilsant l'api V1, j'ai pu bricoler quelque chose en attendant la v2 qui ne fonctionne pas chez moi. Faut que je regarde plus avant...
J'ai aussi tester celui de HA, pensant pousser les data via MQTT qui bridge mes deux apps, mais il ne couvre pas toutes les infos :(

Je vous partage ce que j'ai fait et j'aurais deux questions, n'étant pas sur de bien lire le code

Dans le dossier resources/daemon:
j'ai bricolé un Dockerfile qui fonctionne
```
# derniere version de python (min 3.6 pour la lib wideq)
FROM python:alpine

WORKDIR /usr/src/app

# copy only requirements and install python dependencies
# copy only requirements + wideq coming from plugin installation and install python dependencies
COPY ../wideq requirements.txt ./
RUN pip install --no-cache-dir -r requirements.txt && \
    mkdir -p /usr/src/app/lgthinq/data && \
    ln -s /usr/src/app/lgthinq/data /usr/data

# default port for jeedom
EXPOSE 5025

# launch the user script
CMD [ "python", "srv.py"]
```

et modfié le docker.sh
```
#!/bin/bash
# start docker container wideq server
# require Dockerfile
app="wideqsrvlg"
docker build -t ${app} .
contID=`docker run --detach --rm --publish 127.0.0.1:5025:5025 \
  --volume /var/www/html/plugins/lgthinq/data:/usr/src/app/lgthinq/data:rw \
  --volume /etc/timezone:/etc/timezone:ro \
  --volume /etc/localtime:/etc/localtime:ro \
  --name=${app} ${app} python srv.py -k mylgtinqApiKey -i http://127.0.0.1 -v`
res="$?"
if [ $res -ne 0 ]
then
   echo "Failed to start lgthinq_srv container with exit code ${res}"
else
   sudo chmod o+w /var/www/html/log/lgthinq_srv
   docker container logs --follow $contID 1>> /var/www/html/log/lgthinq_srv 2>&1 &
fi

```

Nota: Image sur base Alpine

A votre connaissance, 
**y a t'il besoin de partager par volume d'autres fichiers que states.json ?**

**En modifiant le fichier WideqManager.class.php, est-ce suffisant pour reprogrammer la commande de lancement du daemon ?**

Merci par avance. Meilleurs voeux pour cette nouvelle année

Et encore merci pour tout le travail effectué

**Comment by pifou25** [comment]
Bonjour, et merci :)  

Désormais Flobul a créé un plugin lgthinq2 compatible pour l'API v2, et il me semble qu'il gère aussi la v1 (rétrocompatible)
Il faut l'installer via la méthode github il n'est pas encore publié (et du coup pas encore de tag sur community):
https://community.jeedom.com/t/probleme-tv-lg-oled55bx6lb-impossible-de-trouver-lg-connect-apps/112923/17
